pin-project-internal/src/utils.rs

```diff
@@ -7,6 +7,10 @@ pub(crate) fn proj_ident(ident: &Ident) -> Ident {
     format_ident!("__{}Projection", ident)
```

Comment:
The way to refer the projected-type should be improved in #46, but should this be `format_ident!("{}Projection", ident)`?
Personally, I don't want to recommend relying on the internal implementation of proc-macro, but I don't plan to change this frequently.

Reply:
cc @Nemo157

Reply:
I only used the type name for the example in #43 because it was the only way to implement it then, I'm not using it in any real code and wouldn't expect it to stay the same across builds (I would even be for hashing the input AST and sticking part of that hash in the generated type names to block anyone actually relying on it with any reliability).

Reply:
Okay, thanks for clarifying.

> I would even be for hashing the input AST and sticking part of that hash in the generated type names to block anyone actually relying on it with any reliability

Unfortunately, I think this is difficult because `#[project]` depends on the current behavior (simple naming). So for now, I will keep the current one.